### PR TITLE
Process configchange event and save the parameters in some data structure.

### LINKED ISSUE #186: Process configchange event to record minimum gas price and gas parameters
Depends on https://github.com/eosnetworkfoundation/eos-evm-contract/issues/679 and https://github.com/eosnetworkfoundation/silkworm/issues/121.

Capture the config change event (see https://github.com/eosnetworkfoundation/eos-evm-contract/issues/679) and store data provided in event (minimum gas price and the five gas parameters) in data associated to each block (mechanism in https://github.com/eosnetworkfoundation/silkworm/issues/121). Do not use those stored values for anything right now. Just use them to verify that the minimum gas price provided in the config change event changes appropriately and is handled properly with forking changes.

Update integration test to verify the node properly processed blocks with a configchange event.

### PR DESCRIPTION
**Pull request by yarkinwho**
Resolve #186 

The planned implementation is like this:

1 ship_receiver_plugin will check if it received a configchange event, if found, make sure it's the first related action in a block, save the new config in the resulting native_block/

2 block_conversion_plugin will make sure if there's a new config in the incoming native_block, it should be the first one native block for a resulting evm block. Then: 
a save the new config in the resulting evm block
b change the evm block's consensus_parameter_index to it's own height.

3 blockchain_plugin will detect if there's new config coming in with the evm blocks. if so:
a save new parameters
b verify_chain for the blocks before if necessary

c insert the incoming blcok
d verify chain (may be not necessary)

-- EDIT: after some double thinking, we do not need this for blockchain_plugin, let's just try a normal process first.

### CODE REVIEW

File: src/block_conversion_plugin.hpp [hunk 1]
```diff
@@ -19,6 +19,17 @@ struct evmtx_v0 {
 using evmtx_type = std::variant<evmtx_v0>;
 EOSIO_REFLECT(evmtx_v0, eos_evm_version, rlpx)
 
+struct gas_parameter_data_v1 {
```
[the remark this answers is not in the document]
Comment (yarkinwho) [reply src/block_conversion_plugin.hpp:22]: Yeah, was copied from there back then. Didn't notice it was changed.

File: src/channels.hpp [hunk 2]
```diff
@@ -38,8 +38,18 @@ namespace channels {
       uint32_t                block_num = 0;
       int64_t                 timestamp = 0;
       uint32_t                lib = 0;
+      std::optional<native_action>  new_config = std::nullopt;
       std::vector<native_trx> transactions;
    };
+
+   struct consensus_parameter_event {
```
[the remark this answers is not in the document]
Comment (yarkinwho) [reply src/channels.hpp]: forgot to delete. deleted

File: src/block_conversion_plugin.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (yarkinwho) [reply src/block_conversion_plugin.hpp]: fixed

File: src/block_conversion_plugin.cpp [hunk 3]
```diff
@@ -267,6 +270,25 @@ class block_conversion_plugin_impl : std::enable_shared_from_this<block_conversi
...
+                     .min_gas_price = std::visit([](auto&& arg) -> auto& { return arg.minimum_gas_price; }, new_config),
+                     .gas_fee_parameters = eosevm::GasFeeParameters {
+                        .gas_txnewaccount = std::visit([](auto&& arg) -> auto& { return arg.gas_parameter.gas_txnewaccount; }, new_config),
+                        .gas_newaccount = std::visit([](auto&& arg) -> auto& { return arg.gas_parameter.gas_newaccount; }, new_config),
+                        .gas_txcreate = std::visit([](auto&& arg) -> auto& { return arg.gas_parameter.gas_txcreate; }, new_config),
+                        .gas_codedeposit = std::visit([](auto&& arg) -> auto& { return arg.gas_parameter.gas_codedeposit; }, new_config),
+                        .gas_sset = std::visit([](auto&& arg) -> auto& { return arg.gas_parameter.gas_sset; }, new_config),
+                     }
+                  });
+                  curr.consensus_parameter_index = curr.header.number;
```
Comment (elmato) [inline src/block_conversion_plugin.cpp:289]: We should save here the consensus parameters to db

File: src/blockchain_plugin.cpp [hunk 4]
```diff
@@ -47,6 +47,35 @@ class blockchain_plugin_impl : std::enable_shared_from_this<blockchain_plugin_im
...
+                        last_consensus_parameters = new_block->consensus_parameters_cache;
+                     }  
+                  }
+                  else {
+                     // Should only reach here if fork happens during version update.
+                     // TODO: Add guards according to eosevm versions.
+                     last_consensus_parameters_index = std::nullopt;
+                     last_consensus_parameters = std::nullopt;
+                  }
+
```
Comment (elmato) [inline src/blockchain_plugin.cpp:78]: We don't need this and the last_consensus_parameters_index / last_consensus_parameters variables

File: src/blockchain_plugin.cpp [hunk 5]
```diff
@@ -88,6 +117,8 @@ class blockchain_plugin_impl : std::enable_shared_from_this<blockchain_plugin_im
       mdbx::env*                                              db_env;
       channels::evm_blocks::channel_type::handle              evm_blocks_subscription;
       std::unique_ptr<ExecutionEngineEx>  exec_engine;
+      std::optional<silkworm::BlockNum>                       last_consensus_parameters_index;
+      std::optional<eosevm::ConsensusParameters>              last_consensus_parameters;
```
Comment (elmato) [inline src/blockchain_plugin.cpp:121]: This is not needed

**Review by elmato — CHANGES_REQUESTED** [review]
~~As a last request, i think we should remove the consensus parameters previously added [here](https://github.com/eosnetworkfoundation/eos-evm-node/pull/190/files#diff-80593411e546176bee31482490117bdb26476e73e12a7b567722ec2b3597aa08R275), when we are reverting each of the forked native blocks [here](https://github.com/eosnetworkfoundation/eos-evm-node/blob/e87222ac61315189300f8751a43ecb3b9fda1e5c/src/block_conversion_plugin.cpp#L237
)~~

We decided to store the hash of the ConsensusParameters values, so there is no need to remove it when a fork happens on the native side

**Review by elmato — APPROVED** [review]

**Pull request merged**